src/lightning/pytorch/strategies/ipu.py

```diff
@@ -23,23 +23,19 @@
 import lightning.pytorch as pl
 from lightning.fabric.plugins import CheckpointIO, ClusterEnvironment
 from lightning.fabric.utilities.cloud_io import get_filesystem
+from lightning.pytorch.accelerators.ipu import _IPU_AVAILABLE, poptorch
 from lightning.pytorch.overrides.base import _LightningModuleWrapperBase
 from lightning.pytorch.plugins.precision import PrecisionPlugin
 from lightning.pytorch.strategies.parallel import ParallelStrategy
 from lightning.pytorch.strategies.strategy import TBroadcast
 from lightning.pytorch.strategies.utils import _fp_to_half
 from lightning.pytorch.trainer.states import RunningStage, TrainerFn
-from lightning.pytorch.utilities import _IPU_AVAILABLE, _POPTORCH_AVAILABLE, rank_zero_warn
+from lightning.pytorch.utilities import rank_zero_warn
 from lightning.pytorch.utilities.data import _get_dataloader_init_args_and_kwargs, _reinstantiate_wrapped_cls
 from lightning.pytorch.utilities.exceptions import MisconfigurationException
 from lightning.pytorch.utilities.model_helpers import is_overridden
 from lightning.pytorch.utilities.types import STEP_OUTPUT
 
-if _POPTORCH_AVAILABLE:
-    import poptorch
-else:
-    poptorch = None
```

Comment:
We still need this for the type annotations in this file. Maybe importing `poptorch` from the ipu file was causing the segfault?

Reply:
I made these changes. The error is still coming when I commit the changes in the [import file](https://github.com/Lightning-AI/lightning/pull/16509/commits/78633ffd624a15c30229ba6e79e22674042c3115).

Reply:
Okay. Let's merge it right now as explore that issue separately